### ISSUE TITLE
Clarification

### ISSUE THREAD

**Issue by AEReed** [abtassociates/eva#401]
Hi,

It appears I'm unable to use Eva because of a high priority issue related to CE fields (status, event, referral). However, since we don't receive CE funding, we are not required to use those fields. I believe there were some changes made previously so being able to assess data quality wasn't limited by particular errors/issues. Will this be adjusted so I can utilize Eva, or should I use the old data quality tool? I know it isn't supported anymore, but if I can't use Eva, I can make due. 

Thanks!

**Comment by AEReed** [comment]
Absolutely! Please see attached. ☺  Also, I had asked our vendor how to resolve this and their solution was filling in the null values…which we can’t really do since we don’t utilize those CE fields for our CE system (currently anyway.)

I hope that helps

Thanks so much!

-April


April Reed
HMIS Application Specialist
MaineHousing
26 Edison Drive
Augusta, ME 04330
(207) 626-4694
Maine Relay 711
[USE THIS LOGO FOR EMAIL SIGNATURE]
www.mainehousing.org<http://www.mainehousing.org/>

**Comment by kiadso** [comment]
Hi, unfortunately whenever you attach a file to an email response, I cannot access that. Can you please upload it directly to this thread on GitHub? https://github.com/abtassociates/eva/issues/401
Thanks

**Comment by AEReed** [comment]
[File-Structure-Analysis-2023-10-16(2).xlsx](https://github.com/abtassociates/eva/files/12920945/File-Structure-Analysis-2023-10-16.2.xlsx)

**Comment by kiadso** [comment]
Thanks again for posting. We will remove the High Priority statuses for these by Thursday, but once Eva is using those data elements in analysis they will be escalated back to High Priority. This will give your vendor time to correct the file structure issues. The good news for your vendor is that version 1.4 of the [HMIS CSV Specifications](https://files.hudexchange.info/resources/documents/HMIS-CSV-Format-Specifications-2024.pdf) has now been released which will make it much clearer how to comply with the specifications, even with missing data. I'll take each one listed in your file separately, including the Errors:

### Event.csv, Event data element

![image](https://github.com/abtassociates/eva/assets/7561522/d9b11d79-8eca-454a-9cf3-a94c9dd2d2fb)

In the HMIS CSV Export Specifications, the Event data element is not set to nullable because without an Event saved, the record is meaningless and can't be used in any analysis. In this case, the vendor should be excluding the record from the Event.csv. They will just need to filter out any records that have nulls in the non-nullable data elements.

### CEParticipation.csv, AccessPoint and ReceivesReferrals data elements

![image](https://github.com/abtassociates/eva/assets/7561522/58bf379c-198f-4456-a931-e75287a224ea)

Similarly to the Event data element, both the AccessPoint and ReceivesReferrals data elements are essential responses to the participation status of the project, so those data elements are set to not nullable in the CSV specifications. In version 1.4 of the specifications, the requirement that there be "at least one record" for each record in Project.csv has been removed so that CE Participation records where either of those elements are null must be excluded from the CEParticipation.csv file. So this is also a case where you vendor will need to address this compliance issue on their side.

### Assessment.csv, AssessmentLocation, AssessmentType, AssessmentLevel, and PrioritizationStatus data elements

![image](https://github.com/abtassociates/eva/assets/7561522/59f1a083-abd6-4b05-a68b-ab88a7f1dd8c)

Eva isn't rejecting your file based on this, and it has been addressed in the updated HMIS CSV specs just today to allow these elements to contain nulls. Eva will be updated Thursday to reflect this change to the specs, and your vendor should not see this File Strucure Analysis Error anymore.

### A note about "Invalid Current Living Situation"

This one appears to be a mapping issue that your vendor will need to fix to avoid incorrect data analysis, once you can use your file in Eva. Eva isn't rejecting the file, but if this isn't fixed, Eva will mis-count the number of clients where CurrentLivingSituation is a permanent housing situation because of the incorrect values here.

tldr: Eva will termporarily accept the misstructured files (try again Thursdasy), but your vendor will need to address each of the issues in your File Structure Analysis _except_ the ones associated with Assessment.csv, since the HMIS CSV Specifications have recently been updated.

**Comment by AEReed** [comment]
Thank you so much! This is all very helpful. I will try again on Thursday/Friday and also pass along this information to our vendor. I do have another unrelated question though please....

One of the reports we were able to upload previously, Eva indicated there were a number of clients with missing UDEs. However, when I checked the client records, none of the UDEs were missing. I noticed they were all clients who have been in the same PSH project for a very long time - 2005, 2007 etc. Is it possible these clients would be getting flagged as errors/missing data because those fields haven't been updated since the clients originally entered the project? 

Thank you for all your help! :)

**Comment by AEReed** [comment]
It was most of them... date of birth, ethnicity, gender, race, length of stay, residence prior, disabling condition, and name quality.

**Comment by kiadso** [comment]
Ok, try everything again Thursday/Friday and enter a new issue if you're still seeing this. You might check your Client.csv file too, to see what was going on. Sometimes your HMIS data isn't pulled into the export correctly. 

Thanks!

**Comment by AEReed** [comment]
Will do! :) Thanks again!
